app/javascript/application.js

```diff
@@ -21,13 +21,15 @@ import 'hint.css/hint.min.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './src/js/axiosWithCsrf';
 import 'controllers';
-import { Turbo } from '@hotwired/turbo-rails';
+import '@hotwired/turbo-rails';
 
 Rails.start();
 
 const componentRequireContext = require.context('./components', true);
 const ReactRailsUJS = require('react_ujs');
 
 ReactRailsUJS.useContext(componentRequireContext);
-
-Turbo.session.drive = false;
+ReactRailsUJS.handleEvent('turbo:load', ReactRailsUJS.handleMount);
+ReactRailsUJS.handleEvent('turbo:before-render', ReactRailsUJS.handleUnmount);
+ReactRailsUJS.handleEvent('turbo:frame-load', ReactRailsUJS.handleMount);
+ReactRailsUJS.handleEvent('turbo:frame-render', ReactRailsUJS.handleUnmount);
```

Comment:
I considered adding a feature flag for this, but based on my manual testing / test suite it doesn't seem to be necessary. Happy to add one if we want to roll out in a safer way